pkg/monitoring/metrics/virt-handler/migrationdomainstats/handler.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ */
+
+package migrationdomainstats
+
+import (
+	"k8s.io/client-go/tools/cache"
+	v1 "kubevirt.io/api/core/v1"
+)
+
+type Handler struct {
+	vmiInformer cache.SharedIndexInformer
+	vmiStats    map[string]*queue
```

Comment:
I think there needs to be a cleanup mechanism.
How many VMIs can we tack here? Where is this going to be cleaned?